### ISSUE TITLE
name may not be null xtext 2.12

### ISSUE THREAD

**Issue by capacman** [eclipse/xtext-extras#145]
2.12 and throws below exception when doing content assist. HDslParsingTest under henkan dsl project can compile successfully. I see this exception when i run my project with RunWebSocketServer and use vscode. Project and associated forum topic is below

https://www.eclipse.org/forums/index.php/t/1086556/
[henkan.dsl.parent.tar.gz](https://github.com/eclipse/xtext-extras/files/1045029/henkan.dsl.parent.tar.gz)

Caused by: java.lang.UnsupportedOperationException: The resource 'file:///home/capacman/programlama/xtext-workspace/henkan.dsl.parent/vscode-hdsl-ext/exampleproject/example.hl' has been loaded without tracking the feature scope. Please check the implementation of IFeatureScopeTracker.Provider.
at org.eclipse.xtext.xbase.typesystem.internal.IFeatureScopeTracker$1.getExpressionScope(IFeatureScopeTracker.java:63)
at org.eclipse.xtext.xbase.typesystem.internal.ResolvedTypes.getExpressionScope(ResolvedTypes.java:1507)
at org.eclipse.xtext.xbase.typesystem.internal.CompoundReentrantTypeResolver.getExpressionScope(CompoundReentrantTypeResolver.java:180)
at org.eclipse.xtext.xbase.typesystem.internal.ForwardingResolvedTypes.getExpressionScope(ForwardingResolvedTypes.java:134)
at org.eclipse.xtext.xbase.ide.contentassist.XbaseIdeContentProposalProvider.createReceiverProposals(XbaseIdeContentProposalProvider.java:606)
at org.eclipse.xtext.xbase.ide.contentassist.XbaseIdeContentProposalProvider.completeBinaryOperation(XbaseIdeContentProposalProvider.java:535)
at org.eclipse.xtext.xbase.ide.contentassist.XbaseIdeContentProposalProvider._createProposals(XbaseIdeContentProposalProvider.java:377)
at org.eclipse.xtext.xbase.ide.contentassist.XbaseIdeContentProposalProvider.createProposals(XbaseIdeContentProposalProvider.java:641)
at org.eclipse.xtext.ide.editor.contentassist.IdeContentProposalProvider.createProposals(IdeContentProposalProvider.java:94)
at org.eclipse.xtext.ide.server.contentassist.ContentAssistService.createProposals(ContentAssistService.java:108)
at org.eclipse.xtext.ide.server.contentassist.ContentAssistService.createCompletionList(ContentAssistService.java:73)
at org.eclipse.xtext.ide.server.LanguageServerImpl.lambda$completion$14(LanguageServerImpl.java:425)
at org.eclipse.xtext.ide.server.WorkspaceManager.doRead(WorkspaceManager.java:234)
at org.eclipse.xtext.ide.server.LanguageServerImpl.completion(LanguageServerImpl.java:427)
at org.eclipse.xtext.ide.server.LanguageServerImpl.lambda$completion$13(LanguageServerImpl.java:406)
at org.eclipse.xtext.ide.server.concurrent.RequestManager.lambda$runRead$2(RequestManager.java:116)
at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
... 5 more

**Comment by cdietrich** [comment]
`org.eclipse.xtext.common.types.access.impl.AbstractRuntimeJvmTypeProvider.TypeInResourceSetAdapter.typeByQueryString`

keeps entries between install/discard derived state
=> proxys are used instead of fresh stuff
=> type is java.lang.Object

**Comment by cdietrich** [comment]
@svenefftinge this is another usecase where the "reuse" of the resourceset does not work as expected

potential workaround: clear the cache

```
import java.lang.reflect.Field;
import java.util.Map;

import org.eclipse.xtext.common.types.JvmIdentifiableElement;
import org.eclipse.xtext.xbase.jvmmodel.JvmElementsProxifyingUnloader;

public class CustomJvmElementsProxifyingUnloader extends JvmElementsProxifyingUnloader {
	
	@Override
	protected void unloadRecursively(JvmIdentifiableElement element) {
		for (Object o : element.eResource().getResourceSet().eAdapters()) {
			if ("TypeInResourceSetAdapter".equals(o.getClass().getSimpleName())) {
				try {
					Field declaredField = o.getClass().getDeclaredField("typeByQueryString");
					if (declaredField != null) {
						declaredField.setAccessible(true);
						Map map = (Map) declaredField.get(o);
						map.clear();
					}
				} catch (NoSuchFieldException | SecurityException | IllegalArgumentException | IllegalAccessException e) {
					e.printStackTrace();
				}
				System.out.println();
			}
		}
		super.unloadRecursively(element);
	}

}
```

```
	def Class<? extends JvmElementsProxifyingUnloader> bindJvmElementsProxifyingUnloader() {
		CustomJvmElementsProxifyingUnloader
	}
```

**Comment by cdietrich** [comment]
fixed in Xtext 2.13
